tech_soln/real_time_analytics.rst

```diff
@@ -0,0 +1,425 @@
+.. _introduction:
+
+Real Time Analytics
+#####################
+
+One workload Citus is commonly used for involves running real-time analytics over large
+streams of data.
+
+For example, say you're a cloud services provider selling a dashboard for HTTP traffic.
+Every time one of your clients receives an HTTP request your service receives a log
+record. You provide your clients insights such as the number of HTTP errors their site
+served. It's important that this data shows up with as little latency as possible, so your
+clients can quickly notice problems with their sites. It's also important that clients are
+able to see graphs of historical trends.
+
+Alternatively, maybe you're building an advertising network and want to show clients
+clickthrough rates on their campaigns. In this example latency is also critical, raw data
+volume is also high, and both historical and live data are important.
+
+In this technical solution we'll demonstrate how to build part of the first example,
+but this architecture would work equally well for the second and many other use-cases.
+
+Running It Yourself
+-------------------
+
+There will be code snippets in this tutorial but they don't specify a complete system.
+There's `a github repo <https://github.com/citusdata/technical-solution-resources>`_ with
+all the details in one place. If you've followed our installation instructions for running
+Citus on either a single or multiple machines you're ready to try it out.
+
+Data Model
+----------
+
+The data we're dealing with is an immutable stream of log data. We'll insert directly into
+Citus but it's also common for this data to first be routed through something like Kafka.
+Doing so has the usual advantages, and makes it easier to pre-aggregate the data once data
+volumes become unmanageably high.
+
+We'll use a simple schema for ingesting HTTP event data. This schema serves as an example
+to demonstrate the overall architecture; a real system might use additional columns.
+
+.. code-block:: sql
+
+  -- this is run on the master
+  CREATE TABLE http_request (
+    site_id INT,
+    ingest_time TIMESTAMPTZ DEFAULT now(),
+
+    url TEXT,
+    request_country TEXT,
+    ip_address INET,
+
+    status_code INT,
+    response_time_msec INT
+  );
+  SELECT master_create_distributed_table('http_request', 'site_id', 'hash');
+  SELECT master_create_worker_shards('http_request', 16, 2);
+
+When we call :ref:`master_create_distributed_table <master_create_distributed_table>`
+we ask Citus to hash-distribute ``http_request`` using the ``site_id`` column. That means
+all the data for a particular site will live in the same shard.
+
+When we call :ref:`master_create_worker_shards <master_create_worker_shards>` we tell
+Citus to create 16 shards and 2 replicas of each shard (for a total of 32 shard replicas).
+:ref:`We recommend <faq_choose_shard_count>` using 2-4x as many shards as CPU cores in
```

Comment:
Minor. What do you think about moving the link to `using 2-4x as many shards <faq_choose_shard_count>`?